### DISCUSSION TITLE
spacy.require_gpu() is true but spacy cli uses CPU

### DISCUSSION THREAD

**Discussion by LarsNeR** [explosion/spaCy#7121] [Help: Other Questions] [answered]
## How to reproduce the behaviour
Use the Quickstart guide on https://spacy.io/usage/training with GPU

I run following code for testing GPU-availability:
`import torch; torch.cuda.is_available() # --> True`
`import spacy; spacy.require_gpu() # --> True`

But when I run `python -m spacy train ./config.cfg ...` like described in the Quickstart it says

"i Using CPU"

Tested that on my local windows machine and on AWS Sagemaker.

## Your Environment

Windows machine:
- **spaCy version:** 3.0.3
- **Platform:** Windows-10-10.0.18362-SP0
- **Python version:** 3.8.6

AWS Sagemaker:
- **spaCy version:** 3.0.3
- **Platform:** Linux-4.14.214-118.339.amzn1.x86_64-x86_64-with-glibc2.9
- **Python version:** 3.6.10

The (important parts of the) config file look like that:
```
[system]
gpu_allocator="pytorch"

[nlp]
pipeline=["transformer", "ner"]

[components]

[components.transformer]
factory="transformer"

[components.transformer.model]
@architectures="spacy-transformers.TransformerModel.v1"
name="bert-base-german-cased"
```

I got the config from https://spacy.io/usage/training "Quickstart" selecting "ner" and "GPU" and I have not changed anything there. After that I did the `init fill-config` and then I tried the `train` resulting in the "i Using CPU"

I expected that when `spacy.require_gpu()` returns `True` the cli command would use the GPU.

**Comment by adrianeboyd** [comment]
This is a good type of question for our new discussions board, so I'll transfer it over there in just a minute. You'll get an automatic message saying this issue is locked, but the discussion can continue on the open thread there.

**Comment by adrianeboyd** [comment, marked as answer]
Whether the GPU is enabled during training is not stored in the config itself, since you might want to use the same config on machines with and without GPUs. There's a command-line flag for `spacy train` to select the GPU, e.g.:

```bash
python -m spacy train ./config.cfg --gpu-id 0
```

**Reply by LarsNeR** [reply]
Thanks that works, my mistake. But maybe that should be mentioned as a side-note at the [Embeddings&Transformers Training](https://spacy.io/usage/embeddings-transformers#transformers-training) or at the [Quickstart](https://spacy.io/usage/training#quickstart)?
